ga4gh/backend.py

```diff
@@ -41,11 +41,27 @@ def _parsePageToken(pageToken, numValues):
 
 def _getVariantSet(request, variantSetIdMap):
     variantSetId = request.variantSetId
+    variantSet = _safeMapQuery(
+        variantSetIdMap, variantSetId,
+        exceptionClass=exceptions.VariantSetNotFoundException)
+    return variantSet
+
+
+def _safeMapQuery(idMap, id_, exceptionClass=None, idErrorString=None):
+    """
+    Attempt to retrieve a value from a map, throw an appropriate error
+    if the key is not present
+    (works for both dict and list types)
```

Comment:
Since slicing syntax is applicable to both dict and list types, _safeMapQuery can be used on either:

`_safeMapQuery({'a':'b'}, 'a')` or `_safeMapQuery(['a'], 0)`